baker/GrapherBaker.tsx

```diff
@@ -206,41 +201,86 @@ const deleteOldGraphers = async (bakedSiteDir: string, newSlugs: string[]) => {
+        minWorkers: 2,
+        maxWorkers: maxWorkers,
+    })
+    const jobs = variableIds.map((variableId) => ({
+        bakedSiteDir: bakedSiteDir,
+        variableId: variableId,
+    }))
+
+    await Promise.all(jobs.map((job) => pool.exec("bakeVariableData", [job])))
+}
+
+export interface BakeSingleGrapherChartArguments {
+    id: number
+    config: string
+    bakedSiteDir: string
+}
+
+export const bakeSingleGrapherChart = async (
+    args: BakeSingleGrapherChartArguments
+) => {
+    const grapher: GrapherInterface = JSON.parse(args.config)
+    grapher.id = args.id
+
+    // Avoid baking paths that have an Explorer redirect.
+    // Redirects take precedence.
+    if (isPathRedirectedToExplorer(`/grapher/${grapher.slug}`)) {
+        console.log(`⏩ ${grapher.slug} redirects to explorer`)
+        return
+    }
+
+    await bakeGrapherPageAndVariablesPngAndSVGIfChanged(
+        args.bakedSiteDir,
+        grapher
+    )
+    console.log(`✅ ${grapher.slug}`)
+}
+
 export const bakeAllChangedGrapherPagesVariablesPngSvgAndDeleteRemovedGraphers =
     async (bakedSiteDir: string) => {
-        const rows: { id: number; config: any }[] = await db.queryMysql(
-            `SELECT id, config FROM charts WHERE JSON_EXTRACT(config, "$.isPublished")=true ORDER BY JSON_EXTRACT(config, "$.slug") ASC`
+        const variablesToBake: { varId: number }[] =
+            await db.queryMysql(`select distinct vars.varID as varId
+            from
+            charts c,
+            json_table(c.config, '$.dimensions[*]' columns (varID integer path '$.variableId') ) as vars
+            where JSON_EXTRACT(c.config, '$.isPublished')=true`)
+
+        await bakeAllPublishedChartsVariableDataAndMetadata(
+            bakedSiteDir,
+            variablesToBake.map((v) => v.varId)
         )
 
-        const newSlugs = []
+        const rows: { id: number; config: string; slug: string }[] =
+            await db.queryMysql(
+                `SELECT id, config, config->>'$.slug' as slug FROM charts WHERE JSON_EXTRACT(config, "$.isPublished")=true ORDER BY JSON_EXTRACT(config, "$.slug") ASC`
+            )
+
+        const newSlugs = rows.map((row) => row.slug)
         await fs.mkdirp(bakedSiteDir + "/grapher")
-        const progressBar = new ProgressBar(
-            "BakeGrapherPageVarPngAndSVGIfChanged [:bar] :current/:total :elapseds :rate/s :etas :name\n",
-            {
-                width: 20,
-                total: rows.length + 1,
-            }
+        const jobs = rows.map((row) => ({
+            id: row.id,
+            config: row.config,
+            bakedSiteDir: bakedSiteDir,
+        }))
+
+        const maxWorkers = MAX_NUM_BAKE_PROCESSES
+        const pool = workerpool.pool(__dirname + "/worker.js", {
+            minWorkers: 2,
+            maxWorkers: maxWorkers,
+        })
+        await Promise.all(
+            jobs.map((job) => pool.exec("bakeSingleGrapherChart", [job]))
```

Comment:
`.then(() => progressBar.tick({ name: job.id }))` here and we can still benefit from the progress bar :)

Of course, displaying the slug would be even better if you want to pass that in, too.

Reply:
Right, nice idea! I was so focused on the two levels of either inside the parallel worker or the overall promise of all jobs that I didn't notice this opportunity, nice catch!